### ISSUE TITLE
TextField handles twinkle when insert text with TextEditingController

### ISSUE THREAD

**Issue by zmtzawqlp** [flutter/flutter#36048]
## Steps to Reproduce
```dart
import 'package:flutter/material.dart';

class TextDeom1 extends StatefulWidget {
  @override
  _TextDeom1State createState() => _TextDeom1State();
}

class _TextDeom1State extends State<TextDeom1> {
  TextEditingController _textEditingController = TextEditingController()
    ..text = "handles twinkle when insert text with TextEditingController";

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Padding(
        padding: EdgeInsets.all(50),
        child: Column(
          children: <Widget>[
            FlatButton(
              child: Text("inset text"),
              onPressed: () {
                insertText("test");
              },
            ),
            Expanded(
                child: TextField(
              controller: _textEditingController,
              maxLines: null,

              //textDirection: TextDirection.rtl,
            ))
          ],
        ),
      ),
    );
  }

  void insertText(String text) {
    var value = _textEditingController.value;
    var start = value.selection.baseOffset;
    var end = value.selection.extentOffset;
    if (value.selection.isValid) {
      String newText = "";
      if (value.selection.isCollapsed) {
        if (end > 0) {
          newText += value.text.substring(0, end);
        }
        newText += text;
        if (value.text.length > end) {
          newText += value.text.substring(end, value.text.length);
        }
      } else {
        newText = value.text.replaceRange(start, end, text);
      }

      setState(() {
        //FocusScope.of(context).requestFocus(_focusNode);
        _textEditingController.value = value.copyWith(
            text: newText,
            selection: value.selection.copyWith(
                baseOffset: end + text.length,
                extentOffset: end + text.length));
      });
    }
  }
}
```

1. click on TextField then TextField handles shows
2. click button to insert text by TextEditingController
3. handles twinkle at Offset.zero
```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, v1.7.8+hotfix.3, on Microsoft Windows [Version 10.0.17134.829], locale zh-CN)

[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[√] Android Studio (version 3.1)
[√] VS Code (version 1.36.0)
[√] Connected device (1 available)
```
handles build before  text(paragraph.dart) is changed,  the selection has changed but text is not changed, so getOffsetForCaret return Offset.zero

**Comment by zmtzawqlp** [comment]
![image](https://user-images.githubusercontent.com/16477333/61114866-d32c8900-a4c3-11e9-8563-41891f702538.png)
_selectionOverlay build before RenderEditable ,so TextPainter is not change at that moment

**Comment by zmtzawqlp** [comment]
more research
there are three  ways to change text
1. paste
2. text input
![image](https://user-images.githubusercontent.com/16477333/61169794-faeb2200-a593-11e9-9962-e9b5d2e55b5b.png)

3.change TextEditingController.value/TextEditingController.text
![image](https://user-images.githubusercontent.com/16477333/61169835-55847e00-a594-11e9-9a8b-b196d14292ce.png)

i can only find in updateEditingValue(text input) method
![image](https://user-images.githubusercontent.com/16477333/61169865-b4e28e00-a594-11e9-8cf7-1ff27aaa9c71.png)
hide overlay when text is changed

but for case 1 and 3, i think also need to hide overlay when text is changed.

hope this is helpful

**Comment by csells** [comment]
@zmtzawqlp what is the behavior you expect?

**Comment by TahaTesser** [comment]
Hi @zmtzawqlp 
Please provide your expect behaviour
Thank you

**Comment by zmtzawqlp** [comment]
It's fixed in 1.17.0 now
